src/org/zaproxy/zap/extension/pscanrulesAlpha/PiiScanner.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @author Michael Kruglos (@michaelkruglos)
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import net.htmlparser.jericho.Source;import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * a scanner to passively scan for the presence of PII in response
+ * Currently only credit card numbers
+ */
+public class PiiScanner extends PluginPassiveScanner {
+
+    /**
+     * Prefix for internationalised messages used by this rule
+     */
+    private static final String MESSAGE_PREFIX = "pscanalpha.piiscanner.";
+
+
+    private static final int PLUGIN_ID = 10062;
+
+    private PassiveScanThread parent = null;
+
+    private static final Pattern AmericanExpress = Pattern.compile("(?:3[47][0-9]{13})");
```

Comment:
These could be an enum?
```Java

private enum CreditCard {
    AMERICAN_EXPRESS("American Express", "(?:3[47][0-9]{13})"),
    // ...
    VISA("Visa", "(?:4[0-9]{12})(?:[0-9]{3})?");

    private final String name;
    private final Pattern pattern;

    CreditCard(String name, String regex) {
        this.name = name;
        this.pattern = Pattern.compile(regex);
    }

    public Matcher matcher(String cc) {
        return pattern.matcher(cc);
    }

    @Override
    public String toString() {
        return name;
    }
}
```
then we could iterate its values (instead of using a map).